### DISCUSSION TITLE
Don't set tag if object already exists

### DISCUSSION THREAD

**Discussion by cliffalbert** [bb-Ricardo/netbox-sync#149] [Ideas]
It looks like netbox-sync also set the netbox-synced tag on objects that already exists.

I am worried that due to this behaviour if you would do a purge OR the referenced object does no longer exist in vcenter the object will be completely removed. This would be a problem primarily for sites where we would also hold other stuff.

I already experienced this when testing netbox-sync on some IP addresses that where already defined in netbox. After the first import they where linked to a VM, but after the first test (with only 1 cluster) I did a purge just for testing purposes and it also removed those IP addresses.

It would be nice to make this optional. A configuration statement like "skip_tag_existing" would this be hard to implement ?

**Comment by bb-Ricardo** [comment]
Hi,

In short: yes, this would be quite difficult, would break the sync.

Long version:
- If an VM is deleted in vCenter, then it doesn't exist anymore, why should it then stay in NetBox?
- If tags are not added to existing object then objects which already exists in NetBox before the first sync will be recreated and you got two versions of it.
- You could consider disabling purge. So objects are not deleted if they gor deleted in vCenter. Would this help?

Also check tis file: https://github.com/bb-Ricardo/netbox-sync/blob/main/module/netbox/object_classes.py

Here are the object classes defined. And only classes with `prune = True` are actually deleted.

Currently this should be:
- Devices
- Device interfaces
- VMs
- VM interfaces
- IP addresse
- Device power ports

Does this help?
